tools/toolchains/xcode_configure/xcode_sdk_frameworks.bzl

```diff
@@ -0,0 +1,141 @@
+    frameworks_dir = sdk_path.get_child("System").get_child("Library").get_child("Frameworks")
+    if not frameworks_dir.exists:
+        # Skip things like DriverKit
+        return []
+    frameworks = frameworks_dir.readdir()
+    return [
+        f.basename.replace(".framework", "")
+        for f in frameworks
+        if f.basename.endswith(".framework") and not f.basename.startswith("_")
+    ]
+
+def _build_file_header():
+    return """load("@build_bazel_rules_swift//swift:swift.bzl", "swift_module_alias", "swift_c_module")
+
+package(default_visibility = ["//visibility:public"])
+        
+"""
+
+IMPORTS_FILE = "bazel_xcode_imports.swift"
+
+def _get_dependency_graph(repository_ctx, developer_dir, sdk_path, target_triple):
+    frameworks = _find_all_frameworks(sdk_path)
+    swift_lib_dir = sdk_path.get_child("usr").get_child("lib").get_child("swift")
+    swift_libs = swift_lib_dir.readdir()
+    frameworks.extend([s.basename.replace(".swiftmodule", "") for s in swift_libs if s.basename.endswith(".swiftmodule") and not s.basename.startswith("_")])
+
+    import_file_content = "".join(["import {}\n".format(f) for f in frameworks])
+    repository_ctx.file(IMPORTS_FILE, content = import_file_content)
+    resource_dir = "{}/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift".format(developer_dir)
+
+    repository_ctx.report_progress("Scanning deps for {}".format(sdk_path))
+    deps_result = repository_ctx.execute(
+        [
+            "swiftc",
+            "-scan-dependencies",
+            "-sdk",
+            sdk_path,
+            "-resource-dir",
+            resource_dir,
+            "-target",
+            target_triple,
+            IMPORTS_FILE,
+        ],
+    )
+    repository_ctx.delete(IMPORTS_FILE)
+    if deps_result.return_code != 0:
+        fail("Could not scan dependencies for {}\n{}".format(sdk_path, deps_result.stderr))
+    return json.decode(deps_result.stdout)
+
+C_MODULE_TMPL = """swift_c_module(
+    name = "{name}_c",
+    module_name = "{name}",
+    system_module_map = "{module_map_file}",
+    deps = [
+{deps}
+    ],
+)
+
+"""
+
+SWIFT_MODULE_TMPL = """swift_module_alias(
+    name = "{name}_swift",
+    deps = [
+{deps}
+    ]
+)
+
+"""
+
+def _sub_bazel_path_vars(path, sdk_path, developer_dir):
+    path = path.replace(str(sdk_path), "__BAZEL_XCODE_SDKROOT__")
+    path = path.replace(str(developer_dir), "__BAZEL_XCODE_DEVELOPER_DIR__")
+    return path
+
+def _rule_for_pkg(developer_dir, sdk_path, pkg):
+    module_path = pkg.get("modulePath", "")
+    if not module_path or IMPORTS_FILE in module_path:
+        return ""
+    module_name = module_path.replace(".pcm", "").replace(".swiftmodule", "")
+
+    pkg_deps = pkg.get("directDependencies", [])
+    deps = ["{}_c".format(d["clang"]) for d in pkg_deps if "clang" in d]
+    deps += ["{}_swift".format(d["swift"]) for d in pkg_deps if "swift" in d]
+    deps_string = "\n".join(sorted(["        \":{}\",".format(d) for d in deps]))
```

Comment:
Sorry if this is a dumb Q but is it expected for all deps be propagated to both `swift_c_module` and `swift_module_alias`? Or should the `_swift` ones be passed only to `swift_module_alias`?

Reply:
Some swift modules depend on c modules; the reverse probably isn't true, but this is really just taking all the dependencies that `swiftc` found and connecting them (by adding the necessary suffix for collisions).

Reply:
k thx for clarifying!